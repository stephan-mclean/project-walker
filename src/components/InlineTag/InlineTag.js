import React from "react";
import PropTypes from "prop-types";
import styles from "./InlineTag.module.scss";

const InlineTag = ({ tag, className, children, ...otherProps }) => {
  const tagClass = `${
    styles.container
  } animate--width flex flex--center tag-${tag}-coloring ${className}`;

  return (
    <div className={tagClass} {...otherProps}>
      <small>{children}</small>
    </div>
  );
};

InlineTag.propTypes = {
  tag: PropTypes.oneOf([
    // TODO: Move this duplicated code
    "one",
    "two",
    "three",
    "four",
    "five",
    "six",
    "seven",
    "eight"
  ]).isRequired,
  className: PropTypes.string,
  children: PropTypes.any
};

export default InlineTag;
