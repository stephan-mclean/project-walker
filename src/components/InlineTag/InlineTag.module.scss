.container {
  height: var(--space-lg);
  width: var(--space-xl);
  padding: 0 var(--component-padding-sm);
  overflow: hidden;
  visibility: hidden;

  animation: animateWidth 0.3s linear;
  animation-fill-mode: forwards;
  transform-origin: 0% 100%;
}

@for $i from 1 to 9 {
  .container:nth-child(#{$i}) {
    animation-delay: $i * 0.125s;
  }
}

@keyframes animateWidth {
  from {
    transform: scaleX(0);
    visibility: hidden;
  }

  to {
    transform: scaleX(1);
    visibility: visible;
  }
}
