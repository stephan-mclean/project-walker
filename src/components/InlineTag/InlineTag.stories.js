import React from "react";
import { storiesOf } from "@storybook/react";

import InlineTag from "./InlineTag";

storiesOf("Components", module).add("InlineTag", () => {
  return (
    <div className="flex">
      <InlineTag tag="one" className="radius-bottom-left" />
      <InlineTag tag="two" className="radius-top-right">
        Some text
      </InlineTag>
    </div>
  );
});
