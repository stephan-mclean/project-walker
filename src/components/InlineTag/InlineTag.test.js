import React from "react";
import { shallow } from "enzyme";

import InlineTag from "./InlineTag";

describe("Testing the InlingTag component", () => {
  it("Should render the correct content", () => {
    const tag = shallow(<InlineTag tag="one">One</InlineTag>);
    expect(tag).toMatchSnapshot();

    expect(tag.text()).toEqual("One");
  });

  it("Should render the correct color class", () => {
    const tagOne = shallow(<InlineTag tag="one" />);
    const tagTwo = shallow(<InlineTag tag="two" />);

    expect(tagOne).toMatchSnapshot();
    expect(tagTwo).toMatchSnapshot();

    expect(tagOne.hasClass("tag-one-coloring")).toEqual(true);
    expect(tagTwo.hasClass("tag-two-coloring")).toEqual(true);
  });

  it("Should allow passing a custom class", () => {
    const tag = shallow(<InlineTag tag="one" className="some-class" />);
    expect(tag).toMatchSnapshot();
    expect(tag.hasClass("some-class")).toEqual(true);
  });
});
