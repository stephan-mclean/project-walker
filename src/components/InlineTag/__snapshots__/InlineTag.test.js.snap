// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the InlingTag component Should allow passing a custom class 1`] = `
<div
  className="container animate--width flex flex--center tag-undefined-coloring some-class"
>
  <small />
</div>
`;

exports[`Testing the InlingTag component Should render the correct color class 1`] = `
<div
  className="container animate--width flex flex--center tag-one-coloring undefined"
>
  <small />
</div>
`;

exports[`Testing the InlingTag component Should render the correct color class 2`] = `
<div
  className="container animate--width flex flex--center tag-two-coloring undefined"
>
  <small />
</div>
`;

exports[`Testing the InlingTag component Should render the correct content 1`] = `
<div
  className="container animate--width flex flex--center tag-one-coloring undefined"
>
  <small>
    One
  </small>
</div>
`;
